### PR TITLE
add_service_db_finalizer_logic

### PR DESCRIPTION
**Pull request by pinikomarov**
Ever since  : https://github.com/openstack-k8s-operators/lib-common/pull/87

Podified operators which use service dbs and the db.CreateOrPatchDB method ,
now will get the finalizer set on the db automatically when using the newest lib-common,
but they won't have the code remove it on cleanup i.e reconcileDelete,
which might make the delete operator action hang.

This adds the service Db removal code in the operator delete (reconcileDelete)
function to work with the above.

Also updated lib-common to latest as to not break lib-common's exported functions

### CODE REVIEW

**Review by steveb — APPROVED** [review]

**Comment by steveb** [comment]
It looks like this change needs to be rebased on master to pick up the pinned operator-lint change

**Comment by pinikomarov** [comment]
ok , rebased

File: go.mod [hunk 1]
```diff
@@ -67,24 +67,24 @@ require (
 	go.uber.org/zap v1.23.0 // indirect
 	golang.org/x/crypto v0.0.0-20220829220503-c86fa9a7ed90 // indirect
 	golang.org/x/net v0.3.1-0.20221206200815-1e63c2f08a10 // indirect
-	golang.org/x/oauth2 v0.0.0-20220909003341-f21342109be1 // indirect
+	golang.org/x/oauth2 v0.2.0 // indirect
```
Comment (steveb) [inline go.mod:70]: Actually I think most of these go.mod changes are unrelated and this one in particular is causing an issue in operator-lint. Can you please rework this so that only lib-common is refreshed to the current commit?

**Review by steveb — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (steveb) [reply go.mod:70]: OK, I think the mod.* changes can be removed from this PR entirely now

**Comment by steveb** [comment]
Re-proposed at https://github.com/openstack-k8s-operators/ironic-operator/pull/59
